crates/runtime/src/lib.rs

```diff
@@ -397,6 +397,16 @@ impl_runtime_apis! {
 
             (messages_to_l1, messages_to_l2)
         }
+
+        fn l1_nonce_unused(nonce: u64) -> bool {
+            StarkFelt::try_from(nonce).map_or_else(
```

Comment:
Also, StarkFelt from u64 is not something that can fail. Any u64 can be stored into a Felt. I think there is another way to achieve this conversion, without the possibility of erroring

Reply:
Regular `StarkFelt::from` works.